dataflux_pytorch/lightning/dataflux_lightning_checkpoint.py

```diff
@@ -0,0 +1,77 @@
+import torch
+from typing import Optional, Dict, Any
+
+from lightning.pytorch.plugins.io import CheckpointIO
+from google.cloud import storage
+from google.api_core.client_info import ClientInfo
+
+class DatafluxLightningCheckpoint(CheckpointIO):
+    """A checkpoint manager for GCS using the :ckass:'CheckpointIO' interface"""
+
+    def __init__(
+        self,
+        project_name: str,
+        bucket_name: str,
+        storage_client: Optional[storage.Client] = None,
+    ):
+        self.project_name = project_name
+        self.bucket_name = bucket_name
+        self.storage_client = storage_client
+        if not storage_client:
+            self.storage_client = storage.Client(
+                project=self.project_name,
+                client_info=ClientInfo(user_agent="dataflux/0.0"),
+            )
+        self.bucket = self.storage_client.bucket(self.bucket_name)
+
+    def _parse_gcs_path(self, path: str) -> str:
```

Comment:
Since the client is initialized with a project name and bucket name, why is this necessary? Could we just have the user provide a path which is relative to the bucket rather than starting with "gcs://<bucket>/"?

Reply:
That was the original implementation but after I wrote the demo/example file I saw that for automated checkpointing this needed to ingest a full path. The full path is what gets passed through to the trainer through the ModelCheckpoint object.

Reply:
Why can't we update the demo to not pass in the "gcs://<bucket>/" part of the path? Alternatively, could we remove the bucket from the constructor in this file? It seems duplicative as-is

Reply:
Yeah it sounds like we are taking on [this feature](https://docs.google.com/document/d/1GVC3bsTGQif03Ps2qFqKRJxxaPFmCcAQoSgL6oRx62o/edit?resourcekey=0-XFc1Vt8Vx4Xl4b8qskVyUQ&tab=t.pxu2qfhdvv07#heading=h.d7bfmv2o8k3y) essentially, which I think it's great to have to make the user experience similar with gcsfs, etc. I'd also agree on just taking away buckets from the constructor. We'd also need work to support this for the two datasets though but definitely for different PRs.

Reply:
I don't see any code changes here, am I missing them? @divrawal can you follow up on it?